### PR TITLE
Add walkthrough for using S3 as a special remote

### PR DESCRIPTION
**Pull request by jsheunis**
@adswa This builds successfully on my mac. I decide to work with a separate simple dataset and not Datalad-101.

Would be good to have you (and/or more people) running through this.

One thing I was uncertain about is when to include term tags and hyperlinks. You'll see that I include them for terms at the start and tend to exclude them later in the doc. Not sure what our approach should be here, i.e. if we should just always include term tags when they are used, or not.

I've run the PNGs through `optipng`.

### CODE REVIEW

**Comment by adswa** [comment]
I will get myself an S3 account and walk through it - thanks a lot all for all your work already!

**Review by adswa — COMMENTED** [review]
I like it, and it works! Thank you! :) I have added a round of comments, with the most severe being a suggestion for using downloadable example data. Let me know what you think. If you don't want to redo the publication but stick to the new example data, I can push a screenshot of the S3 bucket contents

**Comment by adswa** [comment]
> One thing I was uncertain about is when to include term tags and hyperlinks. You'll see that I include them for terms at the start and tend to exclude them later in the doc. Not sure what our approach should be here, i.e. if we should just always include term tags when they are used, or not.

I am not consistent in that either. I, too, tend to add them at the start of the page, but don't repeat them afterwards. At my first read, I didn't spot any terms/hyperlinks I would miss

File: docs/basics/101-139-s3.rst [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+   │   │   └── sub-01_T1w.nii.gz
+   │   └── func
+   │       ├── sub-01_task-rest_run-1_bold.nii.gz
+   │       └── sub-01_task-rest_run-2_bold.nii.gz
+   └── task-rest_bold.json
+
+The first step is to ensure that you have a valid DataLad dataset,
+with ``main`` as the default branch.
+
+.. importantnote:: Ensure main is set as default branch for newly-created repositories
```
Comment (adswa) [inline docs/basics/101-139-s3.rst:91]: #722 adds an FAQ on this that you could link to using
```
 :ref:`some random text of your choice <gitannexdefault>`
```

File: docs/basics/101-139-s3.rst [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+In order to work directly with AWS via your command-line shell, you can
+`install the AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html>`_.
+However, that is not required for this walkthrough.
+
+Lastly, to publish your data repository to GitHub, from which users will be able do install
+the complete dataset, you will need a `GitHub account <https://github.com/join>`_.
+
+Your DataLad dataset
+^^^^^^^^^^^^^^^^^^^^
+For this walkthrough, we are using a basic sample neuroimaging dataset with
```
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:69]: I like the suggestion and it's a great dataset for this use case, thanks!

File: docs/basics/101-139-s3.rst [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+  information. This can be prevented by:
+
+  * `a user/organization setting on GitHub about default branches <https://github.blog/changelog/2020-08-26-set-the-default-branch-for-newly-created-repositories/>`_
+  * setting ``main`` as the default branch by changing your global ``git config``::
+
+      git config --global init.defaultBranch main
+
+If you already have a DataLad dataset, navigate to its root directory. If not, create a
+new directory, navigate to it, copy your data, turn the directory into a DataLad dataset
+with :command:`datalad create --force`, and lastly save the dataset with :command:`datalad save`:
```
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:105]: Very good point, I will update accordingly.

File: docs/basics/101-139-s3.rst [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+   enable :term:`git-annex` to transfer data from and possibly to places that are not Git
+   repositories (e.g., cloud services or external machines such as an HPC
+   system). For example, *s3* special remote uploads and downloads content
+   to AWS S3, *web* special remote downloads files from the web, *datalad-archive*
+   extracts files from the annexed archives, etc. Don’t envision a special-remote
+   as merely a physical place or location – a special-remote is a protocol that
+   defines the underlying transport of your files to and/or from a specific location.
+
+In this section, we provide a walkthrough on how to set up Amazon S3 for hosting
+your DataLad dataset, and how to access this data locally from GitHub.
```
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:26]: Good point. Perhaps it fits best where we mention an Amazon account in the prerequisites. I will find a place for it there.

File: docs/basics/101-139-s3.rst [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+to "Buckets" to see your newly created bucket. It should only have a single 
+``annex-uuid`` file as content, since no actual file content has been pushed yet.
+
+.. figure:: ../artwork/src/aws_s3_bucket_empty.png
+
+   A newly created public S3 bucket
+
+Lastly, for git-annex to be able to download files from the bucket without requiring your
+AWS credentials, it needs to know where to find the bucket. We do this by setting the bucket
+URL, which takes a standard format and can also be copied from your AWS console:
```
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:188]: Indeed. But if it wasn't immediately obvious, I can add some more words to make it obvious.

File: docs/basics/101-139-s3.rst [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+The special remote is ready, and now we want to give people seamless access to the
+DataLad dataset. A common way to do this is to create a sibling of the dataset on
+GitHub using :command:`create-sibling-github`. In order to link the contents in the
+S3 special remote to the GitHub sibling, we also need to configure a publication
+dependency to the ``public-s3`` sibling, which is done with the ``publish-depends <sibling>``
+option. For conistency, we'll give the GitHub sibling the same name as the dataset name. 
+
+.. code-block:: bash
+
+   $ datalad create-sibling-github -d . neuro-data-s3 \
```
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:210]: Actually, I was referring to the name with which the GitHub repo is created, not the name of the sibling known to datalad. Unless my memory is playing trick, I could change this parameter to change the repo name. I'll run this again to make sure, if at least to re-educate myself.

File: docs/basics/101-139-s3.rst (hunk 1 above)
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:91]: Great, thanks!

File: docs/basics/101-139-s3.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (jsheunis) [reply docs/basics/101-139-s3.rst:259]: Indeed, the `-s` flag sets the sibling name, while specifying a name without the `-s` flag sets the github repo name as that name.

**Comment by adswa** [comment]
Sorry, I remembered this PR and that I forgot to check in. I'll build it and review it tomorrow!

**Comment by adswa** [comment]
I think this is great! Thanks A LOT for writing this up! Do you think this is ready to go? (i.e., not WIP anymore?)
@all-contributors please add @jsheunis for content, example

**Comment by jsheunis** [comment]
Yup I think it's ready to go 👍

**Comment by adswa** [comment]
Great! Can you merge the master branch into this branch and push it to fix the remaining conflicts? I can't push to this branch :)

**Pull request merged**